src/api/firebase.ts

```diff
@@ -245,12 +245,15 @@ export async function addItem(
 	});
 }
 
-export async function updateItem() {
-	/**
-	 * TODO: Fill this out so that it uses the correct Firestore function
-	 * to update an existing item. You'll need to figure out what arguments
-	 * this function must accept!
-	 */
+export async function updateItem(listPath: string, item: ListItem) {
+	const itemDocRef = doc(db, listPath, "items", item.id);
+
+	const updates: Partial<ListItem> = {};
```

Comment:
Very nice use of `Partial`! This is comes from a family of types in Typescript know as Utility Types. They're one of my favorite parts about the language. Instead of using `Partial` here, which allows for anything, I think this would be a bit more clear

```
const updates: Pick<ListItem, "totalPurchases" | "dateLastPurchased"> = {
 totalPurchases: item.totalPurchases + 1,
 dateLastPurchased: Timestamp.fromDate(new Date())
}
```

Instead of setting the fields later, we can set them directly. Doing that allows us to have a stronger type. Here I'm using a different Utility Type, `Pick`, to select specific fields from the `ListItem` interface I want to be enforced in this type.

By using partial, other code can't tell which fields are actually meant to be there. `totalPurchases` and `dateLastPurchased` could be `null`, and `dateNextPurchased` might also be there. Less applicable for this use case, but always good to be explicit where you can 😄

Reply:
https://www.typescriptlang.org/docs/handbook/utility-types.html

Reply:
I really like this! I had thought about it being just some of the types we need so when figuring out how to utilize the `ListItem` type again the Partial made sense and most I saw used like an `Omit` I think it was but `Pick` makes so much sense! The setting of the  updates directly is such a good point as well!

Reply:
What a handy and flexible tool!!! My prior understanding of 'Pick' was being able to create a type from another Type model as a means to specify the kind of values your properties will have ("string", "number", etc), but the option to even go further and write blocks of logic as you set the type..is next level.